ext/bz2/bz2.c

```diff
@@ -379,7 +379,7 @@ static PHP_FUNCTION(bzopen)
 		}
 
 		if (CHECK_ZVAL_NULL_PATH(file)) {
-			zend_type_error("Filename must not contain null bytes");
+			zend_argument_value_error(1, "must not contain null bytes");
```

Comment:
or should I change it back to a type error?

Reply:
I think the value error is the proper way to go here, at least from a logical stand point

Reply:
My feelings as well! P.S. null byte errors are 50-50% TypeErrors and ValueErrors in the codebase, so we'll have to adapt them accordingly.

Reply:
I think it is a type error because the ZPP PATH (P modifier) which is used to check for strings without any null bytes so I would imagine that would need to change too in that case.

Reply:
Yes, these are type error because of zpp. And I think it's a type error in that case, because zpp always throws TypeError (even ArgumentCountError is a sub-class of TypeError, despite that not making any sense)

Reply:
Thank you all for the explanation! I've just changed it back to `TypeError`